src/vm/ceeload.cpp

```diff
@@ -3406,6 +3406,67 @@ BOOL Module::IsInCurrentVersionBubble()
 #endif // FEATURE_NATIVE_IMAGE_GENERATION
 }
 
+#if defined(FEATURE_READYTORUN) && !defined(FEATURE_READYTORUN_COMPILER)
+//---------------------------------------------------------------------------------------
+// Check if the target module is in the same version bubble as this one
+// The current implementation uses the presence of an AssemblyRef for the target module's assembly in
+// the native manifest metadata.
+//
+// Arguments:
+//      * target - target module to check
+//
+// Return Value:
+//      TRUE if the target module is in the same version bubble as this one
+//
+BOOL Module::IsInSameVersionBubble(Module *target)
+{
+    CONTRACT(BOOL)
+    {
+        NOTHROW;
+        GC_NOTRIGGER;
+        MODE_ANY;
+    }
+    CONTRACT_END;
+
+    if (this == target)
+    {
+        RETURN TRUE;
+    }
+
+    if (!HasNativeOrReadyToRunImage())
+    {
+        RETURN FALSE;
+    }
+
+    // Check if the current module's image has native manifest metadata, otherwise the current->GetNativeAssemblyImport() asserts.
+    COUNT_T cMeta=0;
+    const void* pMeta = GetFile()->GetOpenedILimage()->GetNativeManifestMetadata(&cMeta);
+    if (pMeta == NULL)
+    {
+        RETURN FALSE;
+    }
+
+    IMDInternalImport* pMdImport = GetNativeAssemblyImport();
```

Comment:
Where is the logic that guarantees that assemblies participating in the version bubble appear in the native manifest  metadata?

Reply:
The Module::ExpandAll iterates over all type defs and refs in the module being crossgen-ed and loads each type using LoadTypeDefOrRefHelper. That ends up calling Module::LoadAssembly for the type and calling RegMeta::DefineAssemblyRef that adds the assembly to a list of assembly refs that are later saved in the native manifest.

Reply:
ok